dataset/dataset_test.go

```diff
@@ -699,6 +699,70 @@ func TestClient_GetDatasetCurrentAndNext(t *testing.T) {
 
 }
 
+func TestClient_GetFullEditionsDetails(t *testing.T) {
+	t.Parallel()
+	Convey("given a 200 status with valid empty body is returned", t, func() {
+		httpClient := createHTTPClientMock(MockedHTTPResponse{http.StatusOK, EditionsDetails{}, nil})
+		datasetClient := newDatasetClient(httpClient)
+
+		Convey("when GetFullEditionsDetails is called", func() {
+			editions, err := datasetClient.GetFullEditionsDetails(ctx, userAuthToken, serviceAuthToken, collectionID, "123")
+
+			Convey("a positive response is returned with empty editions", func() {
+				So(err, ShouldBeNil)
+				So(editions, ShouldResemble, []EditionsDetails(nil))
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkRequestBase(httpClient, http.MethodGet, "/datasets/123/editions", "")
+			})
+		})
+	})
+
+	Convey("given a 200 status with empty body is returned", t, func() {
+		httpClient := createHTTPClientMock(MockedHTTPResponse{http.StatusOK, []byte{}, nil})
+		datasetClient := newDatasetClient(httpClient)
+
+		Convey("when GetFullEditionsDetails is called", func() {
+			_, err := datasetClient.GetFullEditionsDetails(ctx, userAuthToken, serviceAuthToken, collectionID, "123")
+
+			Convey("a positive response is returned", func() {
+				So(err, ShouldNotBeNil)
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkRequestBase(httpClient, http.MethodGet, "/datasets/123/editions", "")
+			})
+		})
+	})
+
+	Convey("given a 200 status with valid body is returned", t, func() {
```

Comment:
Should we be extending these tests to include 404 not found and 500 internal server errors are handled correctly by client?